torch/jit/__init__.py

```diff
@@ -559,6 +564,18 @@ def init_then_register(self, *args, **kwargs):
         return super(ScriptMeta, cls).__init__(name, bases, attrs)
 
 
+class ScriptMethodWrapper(object):
+    def __init__(self, script_method):
+        self.script_method = script_method
+
+    def __call__(self, *args, **kwargs):
+        return self.script_method(*args, **kwargs)
+
+
+def _build_wrap(wrap, script_method):
+    return functools.wraps(wrap)(ScriptMethodWrapper(script_method))
```

Comment:
Is this really necessary? Why can't we just call it on the value returned from `_get_method`?

Reply:
Yes, I think it is necessary, because `torch._C.ScriptMethod` has many read-only :
```python
import torch
import functools

def f():
    """a function"""
    pass

class M(torch.jit.ScriptModule):

    @torch.jit.script_method
    def unchanged(self, x):
        return x

    def forward(self, thing):
        return self.unchanged(thing)

m = M()
functools.wraps(f)(m.unchanged)
```
gives
```
Traceback (most recent call last):
  File "test3.py", line 18, in <module>
    functools.wraps(f)(m.unchanged)
  File "/Users/gaoxiang/anaconda/lib/python3.6/functools.py", line 65, in update_wrapper
    setattr(wrapper, attr, value)
AttributeError: 'torch._C.ScriptMethod' object attribute '__module__' is read-only
```